crates/pallet-feeds/src/lib.rs

```diff
@@ -17,29 +18,78 @@ pub mod pallet {
     pub struct Pallet<T>(_);
 
     pub type PutDataObject = Vec<u8>;
+    pub type FeedId = u64;
+
+    // TODO: make it more generic
+    #[derive(Encode, Decode, Debug, Clone, Eq, PartialEq)]
+    pub struct ObjectMetadata {
+        pub feed_id: FeedId,
+        pub hash: H256,
+        pub number: u32,
+    }
+
+    #[pallet::storage]
+    pub type Feeds<T: Config> = StorageMap<_, Blake2_128Concat, FeedId, (H256, u32), OptionQuery>;
```

Comment:
Consider creating a struct instead of `(H256, u32)`, their scale-encoded representation will be exactly the same, but readability will increase significantly.